lib/AST/NameLookup.cpp

```diff
@@ -594,11 +595,25 @@ static void recordShadowedDecls(ArrayRef<ValueDecl *> decls,
   }
 }
 
-bool swift::removeShadowedDecls(SmallVectorImpl<ValueDecl*> &decls,
-                                const DeclContext *dc) {
+static void
+recordShadowedDecls(ArrayRef<OperatorDecl *> decls, const DeclContext *dc,
+                    llvm::SmallPtrSetImpl<OperatorDecl *> &shadowed) {
+  // Always considered to have the same signature.
+  recordShadowedDeclsAfterSignatureMatch(decls, dc, shadowed);
+}
+
+static void
+recordShadowedDecls(ArrayRef<PrecedenceGroupDecl *> decls,
+                    const DeclContext *dc,
+                    llvm::SmallPtrSetImpl<PrecedenceGroupDecl *> &shadowed) {
+  // Always considered to have the same signature.
+  recordShadowedDeclsAfterSignatureMatch(decls, dc, shadowed);
+}
+
+template <typename T, typename Container>
+static bool removeShadowedDeclsImpl(Container &decls, const DeclContext *dc) {
```

Comment:
`const` please

Reply:
For `decls`? The function mutates the passed vector, so would need to be changed to return a new vector, is this what you meant?

Reply:
Ooh, yikes. Okay, that makes sense.